hathor/nanocontracts/fields/deque_container.py

```diff
@@ -44,88 +45,87 @@ def last_index(self) -> int:
         return self.first_index + self.length - 1
 
 
-_METADATA_NC_TYPE = make_dataclass_opt_nc_type(_DequeMetadata)
+_METADATA_NC_TYPE = make_dataclass_nc_type(_DequeMetadata)
 
 
-class DequeStorageContainer(StorageContainer[Sequence[T]]):
+class DequeContainer(Container[T]):
     # from https://github.com/python/typeshed/blob/main/stdlib/collections/__init__.pyi
-    __slots__ = ('__storage', '__name', '__value', '__metadata_key')
-    __storage: NCContractStorage
-    __name: str
-    __value: NCType[T]
+    __slots__ = ('__storage__', '__prefix__', '__value_node', '__metadata_key')
+    __value_node: ContainerNode[T]
     __metadata_key: bytes
 
-    def __init__(self, storage: NCContractStorage, name: str, value: NCType[T]) -> None:
-        self.__storage = storage
-        self.__name = name
-        self.__value = value
-        self.__metadata_key = f'{name}{KEY_SEPARATOR}{_METADATA_KEY}'.encode()
+    def __init__(self, storage: NCContractStorage, prefix: bytes, value_node: ContainerNode[T]) -> None:
```

Comment:
`prefix` here is the attribute name, right?

Reply:
I guess I like `attr_path` more. And I don't think we need an extra class here. I mean, I guess a separate class wouldn't make your code simpler or safer.